### PR TITLE
Support windows wheels

### PR DESCRIPTION
**Pull request by makslevental**
The pudding is here: https://github.com/Xilinx/llvm-aie/actions/runs/10410089017

image

Looks like all the things we want.

### CODE REVIEW

File: .github/workflows/amd_aie_releases/setup.py
Comment (makslevental) [inline .github/workflows/amd_aie_releases/setup.py:78]: not supported on windows

File: clang/cmake/caches/Peano-AIE-runtime-libraries.cmake
Comment (makslevental) [inline clang/cmake/caches/Peano-AIE-runtime-libraries.cmake:16]: not supported on windows technically but a nice person (@michaelrj-google) showed me the way.

**Review by SagarMaheshwari99 — DISMISSED** [review]

File: clang/cmake/caches/Peano-AIE.cmake
Comment (konstantinschwarz) [inline clang/cmake/caches/Peano-AIE.cmake]: why is this needed?

Comment (makslevental) [reply clang/cmake/caches/Peano-AIE.cmake]: So if IIRC correctly (last week was a long time ago lol), this was needed for `LIBC_TABLEGEN_EXE` to be built. I didn't try to remove it after setting `LLVM_FORCE_BUILD_RUNTIME=ON` so let me try.

Comment (makslevental) [reply clang/cmake/caches/Peano-AIE.cmake]: Cool you were right it was indeed superfluous/redundant. Thanks.

https://github.com/Xilinx/llvm-aie/actions/runs/10459118517

**Review by konstantinschwarz — APPROVED** [review]
I had a look at the libc build system.
Whenever we add libc to LLVM_ENABLE_RUNTIMES, it is also added to LLVM_ENABLE_PROJECTS under the hood to ensure we always find `LIBC_HDRGEN_EXE` or `LIBC_TABLEGEN_EXE`.

Thanks for getting the Windows build up and running!

**Comment by makslevental** [comment]
> Thanks for getting the Windows build up and running!

Well almost... (I should've tested before merging). On windows I'm getting a failure to link when calling `lld` through `clang`:



```
Run:  C:\Users\mlevental\dev_projects\iree-amd-aie\iree-amd-aie\build_tools\llvm-aie\bin\clang.exe C:\Users\mlevental\dev_projects\iree-amd-aie\iree-amd-aie\build_tools\ci\cpu_comparison\cpu_comparison\module_two_matmul_switching_linked_amd_aie_amdaie_xclbin_fb\matmul_small_dispatch_0_matmul_8x8x4_f32_0\input.o \
--target=aie2-none-unknown-elf -Wl,--gc-sections \
-Wl,-T,C:\Users\mlevental\dev_projects\iree-amd-aie\iree-amd-aie\build_tools\ci\cpu_comparison\cpu_comparison\module_two_matmul_switching_linked_amd_aie_amdaie_xclbin_fb\matmul_small_dispatch_0_matmul_8x8x4_f32_0\matmul_small_dispatch_0_matmul_8x8x4_f32_0_core_0_2.elf.ld \
-o C:\Users\mlevental\dev_projects\iree-amd-aie\iree-amd-aie\build_tools\ci\cpu_comparison\cpu_comparison\module_two_matmul_switching_linked_amd_aie_amdaie_xclbin_fb\matmul_small_dispatch_0_matmul_8x8x4_f32_0\matmul_small_dispatch_0_matmul_8x8x4_f32_0_core_0_2.elf -v
```

results in an error because libc/m/crt aren't found/identified correctly:

```
 "C:\\Users\\mlevental\\dev_projects\\iree-amd-aie\\iree-amd-aie\\build_tools\\llvm-aie\\bin\\ld.lld" \
"C:\\Users\\mlevental\\dev_projects\\iree-amd-aie\\iree-amd-aie\\build_tools\\ci\\cpu_comparison\\cpu_comparison\\module_two_matmul_switching_linked_amd_aie_amdaie_xclbin_fb\\matmul_small_dispatch_0_matmul_8x8x4_f32_0\\input.o" \
--gc-sections -T "C:\\Users\\mlevental\\dev_projects\\iree-amd-aie\\iree-amd-aie\\build_tools\\ci\\cpu_comparison\\cpu_comparison\\module_two_matmul_switching_linked_amd_aie_amdaie_xclbin_fb\\matmul_small_dispatch_0_matmul_8x8x4_f32_0\\matmul_small_dispatch_0_matmul_8x8x4_f32_0_core_0_2.elf.ld" \
"-LC:\\Users\\mlevental\\dev_projects\\iree-amd-aie\\iree-amd-aie\\build_tools\\llvm-aie\\lib\\clang\\19\\lib\\aie2-none-unknown-elf" \
"C:\\Users\\mlevental\\dev_projects\\iree-amd-aie\\iree-amd-aie\\build_tools\\llvm-aie\\lib\\clang\\19\\lib\\aie2-none-unknown-elf\\libclang_rt.builtins.a" \
-lc -lm crt0.o crt1.o -o \
"C:\\Users\\mlevental\\dev_projects\\iree-amd-aie\\iree-amd-aie\\build_tools\\ci\\cpu_comparison\\cpu_comparison\\module_two_matmul_switching_linked_amd_aie_amdaie_xclbin_fb\\matmul_small_dispatch_0_matmul_8x8x4_f32_0\\matmul_small_dispatch_0_matmul_8x8x4_f32_0_core_0_2.elf"
ld.lld: error: unable to find library -lc
ld.lld: error: unable to find library -lm
ld.lld: error: cannot open crt0.o: no such file or directory
ld.lld: error: cannot open crt1.o: no such file or directory
```

I can add `-Wl,-Lllvm-aie\lib` and this fixes the libc/m problem but not the crt problem. Any ideas @konstantinschwarz

**Comment by makslevental** [comment]
I took a look at the linux wheel and those are all under `lib/aie2-none-unknown-elf` (as I guess is expected). No clue why on windows they get shoved into just lib - maybe something having to do with `LLVM_FORCE_ENABLE_RUNTIME`?

Actually come to think of it, I wonder if these are even the correct things (notice the absence of `aie2-none-unknown-elf`:

image

although there is a libclang_rt:

image

**Comment by jhuber6** [comment]
> I had a look at the libc build system. Whenever we add libc to LLVM_ENABLE_RUNTIMES, it is also added to LLVM_ENABLE_PROJECTS under the hood to ensure we always find `LIBC_HDRGEN_EXE` or `LIBC_TABLEGEN_EXE`.
> 
> Thanks for getting the Windows build up and running!

This shouldn't be required with the "new headergen", however there are still a few things in place that need to be trimmed up. The new approach just uses `pyyaml` instead of a dedicated TableGen utility.

**Comment by makslevental** [comment]
> This shouldn't be required with the "new headergen", however there are still a few things in place that need to be trimmed up. The new approach just uses `pyyaml` instead of a dedicated TableGen utility.

@michaelrj-google landed this https://github.com/llvm/llvm-project/pull/104629 to unblock me on the headergen issue but over here we're a little ways behind HEAD 😅

**Pull request merged**
